test/lib/server/services/run/RunService.test.js

```diff
@@ -310,4 +310,137 @@ module.exports = () => {
             { runPatch: { runQuality: RunQualities.TEST } },
         );
     });
+
+    it('should successfully update run with eorReasons with reasonTypeId', async () => {
+        const runNumber = 1;
+
+        const eorReasons = [
+            {
+                reasonTypeId: 1,
+                description: 'test1',
+            },
+            {
+                reasonTypeId: 1,
+                description: 'test2',
+            },
+        ];
+
+        const run = await runService.update(
+            { runNumber },
+            { relations: { eorReasons: eorReasons } },
+        );
+
+        expect(run.eorReasons).to.lengthOf(2);
+        // Description is unique, should be equal
+        for (let i = 0; i < run.eorReasons.length; i++) {
+            expect(run.eorReasons[i].description).to.equal(eorReasons[i].description);
```

Comment:
Why don't you also test the reasonTypeId here?

Reply:
Because the reasonTypeId is the same as the first object I am trying to add, the description is more unique here and that way you can be sure that it is the object I am referring to and that not the first object is added twice for example and the second one is not added. That is why I chose for the description, but if you would rather test the resaonTypeId than I can change it to that if you want.

Reply:
Thank you for the explanation, I have implemented this now in the tests as you requested.